ledgers/ccf/scripts/start_ccf_network.sh

```diff
@@ -65,41 +78,71 @@ then
     exit -1
 fi
 
+# -----------------------------------------------------------------
+# clean up any old ledger keys
+# -----------------------------------------------------------------
 rm -f ${PDO_LEDGER_KEY_ROOT}/networkcert.pem
 rm -f ${PDO_LEDGER_KEY_ROOT}/ledger_authority_pub.pem
 
 # ensure that we use an IP address for the interface
 F_INTERFACE_ADDRESS=$(force_to_ip ${F_INTERFACE})
-
 yell start ccf network with "local://${F_INTERFACE_ADDRESS}:${F_PORT}"
 
+# -----------------------------------------------------------------
+# run the CCF script to start the network. this script must operate
+# in the python virtual environment created for it in ${F_CCF_LEDGER_DIR}
+# note that simply executing python out of the ccf ledger virtual env
+# is not sufficient since ccf code has expectations for the PATH which
+# is only changed when you activate.
+# -----------------------------------------------------------------
 say attempt to start ccf node
-${CCF_BASE}/bin/sandbox.sh \
-    --verbose \
-    --host-log-level info \
-    -p ${PDO_HOME}/ccf/lib/libpdoenc \
-    -n "local://${F_INTERFACE_ADDRESS}:${F_PORT}" \
-    -e virtual -t virtual \
-    --workspace ${PDO_HOME}/ccf/workspace \
-    --initial-node-cert-validity-days 365 \
-    --initial-service-cert-validity-days 365 &
-
-sleep 5
-
-while [ ! -f ${PDO_HOME}/ccf/workspace/sandbox_common/service_cert.pem ]; do
+
+source ${F_CCF_LEDGER_DIR}/bin/activate
+CURL_CLIENT=ON INITIAL_MEMBER_COUNT=1 \
```

Comment:
Probably good to reference also the source of below?
```suggestion
# Note below is adopted from (the end of ) CCF's tests/sandbox/sandbox.sh ..
CURL_CLIENT=ON INITIAL_MEMBER_COUNT=1 \
```

BTW: just noticed that even though it's a test single-node installment, seems we (already in previous version0 foresaw it for long-term deployment given that we bumped up certs lifetime from 90 to 365 days ;-)

Reply:
i don't think we were that intentional about cert updates. there is a big open question about multiple-node ledgers. the plumbing is there... but the implementation does not exist.